src/vcf/record.cpp

```diff
@@ -340,20 +369,30 @@ namespace ebi
             if (allele == ".") { continue; } // No need to check missing alleles
 
             // Discard non-integer numbers
-            if (std::find_if_not(allele.begin(), allele.end(), isdigit) != allele.end()) {
-                throw new SamplesFieldBodyError{line, "Allele index " + allele + " is not an integer number",
-                                                "GT", ploidy};
-            }
+            check_samples_alleles_int(allele, ploidy);
```

Comment:
I would favor clearer (longer) method names like "check_samples_allele_is_integer" and remove the comments like "discard non-integer numbers"